prov/gni/include/gnix_vector.h

```diff
@@ -86,7 +87,7 @@ struct gnix_vector;
 
 struct gnix_vector_iter {
 	struct gnix_vector *vec;
-	int cur_idx;
+	uint64_t cur_idx;
```

Comment:
The vector can be as large as the maximum value a uint64_t can hold.  I changed this so that the iterator idx doesn't overflow if the vector the iterator is operating on is very large.

 Although, a vector of the max size a uint32_t can hold is 2^32 - 1 elements which uses ~35GB of memory -- i think I'll change this to a uint32_t as well as the fields in the vector handle.



On Fri, Jul 1, 2016 at 7:08 PM -0500, "Sung-Eun Choi" <notifications@github.com<mailto:notifications@github.com>> wrote:


In prov/gni/include/gnix_vector.h<https://github.com/ofi-cray/libfabric-cray/pull/887#discussion_r69368207>:

> @@ -86,7 +87,7 @@ struct gnix_vector;
>
>  struct gnix_vector_iter {
>       struct gnix_vector *vec;
> -     int cur_idx;
> +     uint64_t cur_idx;


I can't quite figure out why this has changed to uint64_t.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ofi-cray/libfabric-cray/pull/887/files/9f4cf047cb13c40c8fcd09da66360d86d60b795f#r69368207>, or mute the thread<https://github.com/notifications/unsubscribe/AHET8qCrGPygQW6KKv-W3UZH5rYcR5-eks5qRavdgaJpZM4JChQw>.